### ISSUE TITLE
nested higher-order functions

### ISSUE THREAD

**Issue by Khabib73** [wemake-services/wemake-python-styleguide#3693]
Hi! @sobolevn 

What do you think about this code:

```python
res = reduce(
    lambda acc, x: acc + x,
    filter(
        lambda x: x > 5,
        map(
            lambda x: x + 1,
            map(
                lambda x: x * 2,
                filter(lambda x: x % 2 == 0, numbers)
            )
        )
    ),
    0
)
```

How about adding rules to simplify nesting? 

It's should be like this:

```python

step1 = filter(lambda x: x % 2 == 0, numbers)
step2 = map(lambda x: x * 2, step1)
step3 = map(lambda x: x + 1, step2)
step4 = filter(lambda x: x > 5, step3)
```
please, assign me , if you think it should be a violation :)

**Comment by sobolevn** [comment]
We can count the `lambda` nesting level. I think that allowing 2 levels by default is ok :)
But, we should count not a single nested `lambda` level, but layers of `lambda` nesting, as you show in the original example.
